lib/provider.go

```diff
@@ -307,19 +307,18 @@ func (p *Provider) roleSessionName() string {
 	return fmt.Sprintf("%d", time.Now().UTC().UnixNano())
 }
 
-func (p *Provider) GetRoleARN() (string, error) {
-	creds, err := p.getSamlSessionCreds()
-	if err != nil {
-		return "", err
-	}
+// GetRoleARN makes a call to AWS to get-caller-identity and returns the
+// assumed role's name and ARN.
+func (p *Provider) GetRoleARN(c *credentials.Value) (string, error) {
```

Comment:
We should avoid breaking API, but considering this method is currently broken, I don't mind.

Also, since this method doesn't use `p`, maybe let's just make it a plain function.